pkg/engine/plan.go

```diff
@@ -193,17 +193,19 @@ func (res *planResult) Chdir() (func(), error) {
 // resulting Snapshot, no matter whether an error occurs or not; an error, if something went wrong; the step that
 // failed, if the error is non-nil; and finally the state of the resource modified in the failing step.
 func (res *planResult) Walk(cancelCtx *Context, events deploy.Events, preview bool) (deploy.PlanSummary, error) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
 	opts := deploy.Options{
+		Context:  ctx,
```

Comment:
How about just passing this as an argument to `Iterate` like we do with `NewPlanExecutor`?